rye/src/lock.rs

```diff
@@ -194,6 +197,21 @@ pub fn update_workspace_lockfile(
     Ok(())
 }
 
+fn maybe_write_requirements_to_temp(requirements: &HashSet<Requirement>) -> Result<Option<NamedTempFile>, Error> {
+    let over_temp_file: Option<NamedTempFile> = match requirements.is_empty() {
```

Comment:
If this prefers `if` instead of `match`  (I don't mind match bool, but if is less surprising) and removes the over_temp_file binding, the whole function body is just the if-else, then this function becomes simple.

Reply:
good point